DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -560,6 +564,85 @@ bool muon::isGoodMuon( const reco::Muon& muon,
     if ( nMatch >= minNumberOfMatches ) return true;
     else return false;
   } // RPCMu
+    
+  if ( type == ME0Mu )
+  {
+    if ( minNumberOfMatches == 0 ) return true;
+    
+    int nMatch = 0;
+    for ( std::vector<reco::MuonChamberMatch>::const_iterator chamberMatch = muon.matches().begin();
+          chamberMatch != muon.matches().end(); ++chamberMatch )
+    {
+      if ( chamberMatch->detector() != 5 ) continue;
+
+      const double trkX = chamberMatch->x;
+      const double errX = chamberMatch->xErr;
+      const double trkY = chamberMatch->y;
+      const double errY = chamberMatch->yErr;
+
+      for ( std::vector<reco::MuonSegmentMatch>::const_iterator segment = chamberMatch->me0Matches.begin(); segment != chamberMatch->me0Matches.end(); ++segment )
+      {
+          
+        const double me0X = segment->x;
+        const double me0ErrX = segment->xErr;
+        const double me0Y = segment->y;
+        const double me0ErrY = segment->yErr;
+
+        const double dX   = fabs(me0X - trkX);
+        const double dY   = fabs(me0Y - trkY);
+        const double pullX   = dX/std::sqrt(errX + me0ErrX);
+        const double pullY   = dY/std::sqrt(errY + me0ErrY);
+          
+        if ( (dX < maxAbsDx or pullX < maxAbsPullX) and (dY < maxAbsDy or pullY < maxAbsPullY) )
+        {
+          ++nMatch;
+          break;
+        }
+      }
+    }
+
+    if ( nMatch >= minNumberOfMatches ) return true;
+    else return false;
+  } // ME0Mu
+  if ( type == GEMMu )
```

Comment:
actually I am not sure if this is really needed, because gemMuons won't be used standalone. Anyway I added a new function because it is not said the two will be the same in the future. I don't think it is the case to spend time on this now, if this can not remain as it is, I will remove it.